### PR TITLE
[DROOLS-5688] Implement public API

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @mariofusco @jiripetrlik

See https://issues.redhat.com/browse/DROOLS-5688

This PR adapt codegen/evaluation to new Public API.

It depends on https://github.com/kiegroup/drools/pull/3162

### CODE REVIEW

**Comment by gitgabrio** [comment]
Jenkins run cdb

**Review by danielezonca — APPROVED** [review]
LGTM

File: kogito-build-parent/pom.xml [hunk 1]
```diff
@@ -130,7 +130,7 @@
     <version.org.keycloak>11.0.0</version.org.keycloak>
     <version.org.mockito>3.3.3</version.org.mockito>
     <version.org.mvel>2.4.7.Final</version.org.mvel>
-    <version.org.kie7>7.45.0.t20201014</version.org.kie7>
+    <version.org.kie7>7.45.0-SNAPSHOT</version.org.kie7>
```
[the remark this answers is not in the document]
Comment (gitgabrio) [reply kogito-build-parent/pom.xml:133]: @danielezonca 
Yup - I had to because tag release is missing the public api

**Review by jiripetrlik — APPROVED** [review]
LGTM.

**Comment by triceo** [comment]
@gitgabrio 7.45.0.Final is out - can we merge this?

**Comment by mariofusco** [comment]
Included and replaced by https://github.com/kiegroup/kogito-runtimes/pull/852

**Pull request closed without merging**